pkg/apis/crd.projectcalico.org/v1/ippool.go

```diff
@@ -54,13 +65,35 @@ type IPPoolSpec struct {
 	// When disabled is true, Calico IPAM will not assign addresses from this pool.
 	Disabled bool `json:"disabled,omitempty"`
 
-	// The block size to use for IP address assignments from this pool. Defaults to 26 for IPv4 and 112 for IPv6.
+	// Disable exporting routes from this IP Pool's CIDR over BGP. [Default: false]
+	DisableBGPExport bool `json:"disableBGPExport,omitempty" validate:"omitempty"`
+
+	// The block size to use for IP address assignments from this pool. Defaults to 26 for IPv4 and 122 for IPv6.
 	BlockSize int `json:"blockSize,omitempty"`
 
 	// Allows IPPool to allocate for a specific node by label selector.
 	NodeSelector string `json:"nodeSelector,omitempty" validate:"omitempty,selector"`
+
+	// Deprecated: this field is only used for APIv1 backwards compatibility.
+	// Setting this field is not allowed, this field is for internal use only.
+	IPIP *IPIPConfiguration `json:"ipip,omitempty" validate:"omitempty,mustBeNil"`
```

Comment:
IMO we shouldn't be including these fields (IPIP / NatOutgoingV1 / AllowedUses) - they're not used by the operator code at all, and are deprecated internal fields from a past version of the API.

Reply:
Good point, I've rolled back the unnecessary changes and only left the `disableBGPExport` field on IPPoolSpec that we need: d28a2184260c1475ae0fa66f766d082a5a4dc2ba